File: fc-chat/demo/part3/ch3/03-2-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/WriteMessage.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;

public class WriteMessage extends BaseRequest {

  private final ChannelId channelId;
  private final String content;

  @JsonCreator
  public WriteMessage(
      @JsonProperty("channelId") ChannelId channelId, @JsonProperty("content") String content) {
    super(MessageType.WRITE_MESSAGE);
    this.channelId = channelId;
    this.content = content;
  }

  public ChannelId getChannelId() {
    return channelId;
  }

  public String getContent() {
    return content;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/WriteMessage.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId


class WriteMessage @JsonCreator constructor(
    @JsonProperty("channelId") val channelId: ChannelId,
    @JsonProperty(
        "content"
    ) val content: String
) :
    BaseRequest(MessageType.WRITE_MESSAGE) {

    }
